Этап 2 Дипломный проект

polyakovanan opened the pull request:
Реализация основной части сервиса для работы с событиями

polyakovanan committed:
Этап 2. Основной сервис
Нулевая итерация основного сервиса. Реализован весь функционал и приложение запускается.
Хоть какой-то проверки работоспособности еще не было

polyakovanan committed:
Этап 2. Основной сервис
Рабочая итерация основного сервиса. Приложение прошло тестирование на тестах постмана для самого приложения и для сервиса статистики.
Осталось покрытие тестами

polyakovanan committed:
Этап 2 Дипломный проект.
Покрытие тестами основного сервиса >85%. Фикс проблемы, из-за которой не всегда правильно собиралась статистика посещений страниц с событиями

polyakovanan committed:
Этап 2 Дипломный проект.
Небольшая уборка в тестах

polyakovanan committed:
Этап 2 Дипломный проект.
Поправил тесты клиента статистики под новый вариант вызова эндпоинта

polyakovanan committed:
Этап 2 Дипломный проект.
Правка чек стайла

polyakovanan committed:
Этап 2 Дипломный проект.
Исправление pom'ника для билда

polyakovanan committed:
Этап 2 Дипломный проект.
Сократил стек трейс, чтобы он влезал в логи в гите. Иначе вообще непонятно, почему там падает, а локально нет

polyakovanan committed:
Этап 2 Дипломный проект.
Убрал стектрейс из ошибки

polyakovanan committed:
+ чекстайл

polyakovanan committed:
+ тесты

polyakovanan committed:
Проверка теории с датой в запросе

polyakovanan committed:
Еще раз

polyakovanan committed:
Еще раз, теперь через CAST

polyakovanan committed:
Должно исправить коннект до сервиса статистики

polyakovanan committed:
А нет, не должно

polyakovanan committed:
Должно решить проблему с lower(bytea)

polyakovanan committed:
Заменил тип данных на TEXT

polyakovanan committed:
Должно решить проблему коннекта до сервиса статистики

polyakovanan committed:
Попытка обойти хитрый баг хибернейта

polyakovanan committed:
Вернул нормальный стектрейс в ошибки и правильные типы данных в БД

Umka7389 commented on ewm-service/src/main/java/ru/practicum/core/persistance/model/ParticipationRequest.java:
```diff
@@ -0,0 +1,32 @@
+public class ParticipationRequest {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Модификаторы доступа не нужны?

polyakovanan replied:
Добавил модификатор private везде, где его не хватало в классах моделей

Umka7389 commented on ewm-service/src/main/java/ru/practicum/core/service/impl/CompilationServiceImpl.java:
```diff
@@ -0,0 +1,87 @@
+import java.util.List;
+import java.util.Set;
+
+@Service("compilationService")
```
Совсем необязательно задавать кастомное имя бину, если у вас одна имплементация

polyakovanan replied:
Убрал явное присвоение имени всем бинам сервисам

Umka7389 reviewed (commented):
Молодец, отличная работа! Простой, понятный код, оставил небольшие рекомендации

polyakovanan committed:
Этап 2 дипломный проект
Добавил недостающие модификаторы доступа в классы моделей
Убрал явное присвоение имен бинам сервисов

Umka7389 reviewed (commented):
Молодец, отличная работа!
